grid2op/tests/test_alert.py

```diff
@@ -13,45 +13,1234 @@
+                            opponent_attack_cooldown=0, 
+                            opponent_attack_duration=99999, 
+                            opponent_budget_per_ts=1000, 
+                            opponent_init_budget=10000., 
+                            opponent_action_class=PlayableAction, 
+                            opponent_class=TestOpponentMultiLines, 
+                            kwargs_opponent=kwargs_opponent,
+                            reward_class=AlertReward(reward_end_episode_bonus=42)
+            ) as env : 
+            env.seed(0)
+            env.reset()
+            for i in range(env.max_episode_duration()):
+                act = self.do_nothing
+                if i == 4 :
+                    act = self.env.action_space({"raise_alert": [1]})
+                elif i == 6 : 
+                    act = self.blackout_action
+                obs, reward, done, info = env.step(act)
+
+                if i in [2,5] : 
+                    assert env._oppSpace.last_attack is not None
+                
+                if i == 3 : 
+                    assert reward == 1
+                elif i == 6 : 
+                    assert reward == 2
+                else : 
+                    assert reward == 0
+            
+                if done : 
+                    break
+
+# return 0 
+    def test_assistant_reward_value_blackout_no_attack_alert(self) -> None :
+
+        """Even if there is a blackout, an we raise an alert
+           we expect a reward of 0 because there is no attack"""
+        with make(
+            self.env_nm,
+            test=True,
+            difficulty="1",
+            reward_class=AlertReward(reward_end_episode_bonus=42)
+        ) as env:
+            env.seed(0)
+            env.reset()
+
+            done = False
+            for i in range(env.max_episode_duration()):
+                act = self.do_nothing
+                if i == 3 : 
+                    act = self.blackout_action
+                elif i == 1:
+                    act = self.env.action_space({"raise_alert": [0]})
+                obs, reward, done, info = env.step(act)
+                if env._oppSpace.last_attack is None : 
+                    assert reward == 0
+                else : 
+                    raise Grid2OpException('No attack expected')
+
+                if done : 
+                    break
+            
+            assert done
+
+# return 0 
+    def test_assistant_reward_value_blackout_no_attack_no_alert(self) -> None :
+        """Even if there is a blackout, an we don't raise an alert
+           we expect a reward of 0 because there is no attack"""
+        with make(
+            self.env_nm,
+            test=True,
+            difficulty="1",
+            reward_class=AlertReward(reward_end_episode_bonus=42)
+        ) as env:
+            env.seed(0)
+            env.reset()
+
+            done = False
+            for i in range(env.max_episode_duration()):
```

Comment:
can you remove this "for" loop and make the thing sequential (you only need to do 3 actions, 3 steps and 3 checks, it's clearer without the for loop I think)